### PR TITLE
Upgrading Ember to version 1.9.0 beta 3

### LINKED ISSUE #1069: Update Wanted: Ember
The app is on Ember 1.6.1, the latest is 1.9.0-beta.3. Additionally, we're on ember-data#1.0.0-beta.7 (latest is 1.0.0-beta.11). I believe the latter has some breaking changes.

Anyone up for this? :)

### PR DESCRIPTION
**Pull request by phillipsj**
This pull request resolves #1069.  Updated Ember and its dependencies and correct all errors that were being received using Ember Inspector.

### CODE REVIEW

File: examples/emberjs/js/controllers/todos_controller.js [hunk 1]
```diff
@@ -22,12 +22,14 @@
 
 				// Clear the "New Todo" text field
 				this.set('newTitle', '');
+                return false; // Returning false to prevent event bubbling
```
Comment (nikolas) [inline examples/emberjs/js/controllers/todos_controller.js]: It's better not to mix tabs and spaces.

File: examples/emberjs/bower.json [hunk 2]
```diff
@@ -5,8 +5,11 @@
     "todomvc-common": "~0.3.0",
     "jquery": "~2.1.0",
     "handlebars": "~1.3.0",
-    "ember": "~1.6.0",
-    "ember-data": "1.0.0-beta.7",
+    "ember": "~1.9.0-beta.3",
```
Comment (nikolas) [inline examples/emberjs/bower.json]: This could now be changed to 1.9.1

File: examples/emberjs/js/controllers/todo_controller.js [hunk 3]
```diff
@@ -33,23 +34,25 @@
 				// Re-set our newly edited title to persist its trimmed version
 				this.set('bufferedTitle', bufferedTitle);
 				this.set('isEditing', false);
```
Comment (nikolas) [inline examples/emberjs/js/controllers/todo_controller.js]: These lines can be changed to:
```js
this.setProperties({
    'bufferedTitle', bufferedTitle,
    'isEditing', false
});
```

File: examples/emberjs/js/controllers/todo_controller.js [hunk 4]
```diff
@@ -33,23 +34,25 @@
...
+                return false;
 			},
 
 			cancelEditing: function () {
 				this.set('bufferedTitle', this.get('title'));
 				this.set('isEditing', false);
```
Comment (nikolas) [inline examples/emberjs/js/controllers/todo_controller.js]: Same here (`setProperties` is preferable to multiple `set` calls).

File: examples/emberjs/js/controllers/todo_controller.js [hunk 5]
```diff
@@ -33,23 +34,25 @@
...
+                return false;
 			},
 
 			removeTodo: function () {
 				this.removeTodo();
+                return false;
```
Comment (nikolas) [inline examples/emberjs/js/controllers/todo_controller.js]: @sindresorhus @phillipsj returning false from an action prevents any parent controllers from handling the action. By default, I leave that out of my actions, and then add it in if necessary. It should be unnecessary here, but I haven't run this code yet

Comment (nikolas) [inline examples/emberjs/js/controllers/todo_controller.js]: Actually, I was wrong. Event bubbling is disabled by default for actions, so these `return false;` statements are definitely unnecessary.

From http://emberjs.com/api/classes/Ember.ObjectController.html#toc_bubbling:

> By default, an action will stop bubbling once a handler defined on the actions hash handles it. To continue bubbling the action, you must return true from the handler

Comment (phillipsj) [inline examples/emberjs/js/controllers/todo_controller.js]: I will get these changes done and pushed up hopefully this week.  I have
been too busy to dedicate the time.  I apologize.

**Comment by phillipsj** [comment]
Okay, thanks everyone for being patient with me.  Hopefully, I took care of all the issues.  If you want I can close this one and open a new one as I went ahead and used ember 1.9.1.

**Comment by passy** [comment]
Thanks! Would you mind squashing the commits before we merge it?

**Comment by sindresorhus** [comment]
> As I said earlier, any files not required to run the app should be removed. That means dotfiles, readmes, unused jquery source files, etc.

Still unresolved.

**Comment by nikolas** [comment]
How do other TodoMVC implementations deal with that? npm installs all these extra files along with the source, and it looks like the other implementations use npm for dependencies.

**Comment by sindresorhus** [comment]
@nikolas They put them in a local `.gitignore`.

**Comment by zzarcon** [comment]
Uff this is crazy... +170K lines for make an upgrade of a minor version :laughing:  :gun:

**Comment by nikolas** [comment]
@zzarcon it's a completely understandable mistake. @phillipsj just needs to add a bunch of files to the `.gitignore`. It's an understandable mistake because I haven't worked on a project that uses npm to handle package management, and then use the `.gitignore` to ignore all the non-essential files in those packages.

**Comment by phillipsj** [comment]
If you think this is crazy and not trying to be helpful, then please by all
means do not accept this request and I will spend what little extra time
that I have contributing to other projects that are more accepting.  Most
have been great and I think each one of you that have been understanding.

**Comment by sindresorhus** [comment]
@phillipsj, just ignore @zzarcon. He's not a part of this project.

We would be happy to accept this PR when the `.gitignore` is sorted out :)

**Comment by zzarcon** [comment]
@phillipsj please apologize me for my comment, 
I'm sorry if you unterstand that, this wasn't my intention. What I'm was trying to say it's that is a bit crazy the number of changes beetwen dependencies; but this isn't your fault of course, so if my comment has ofend you, only forget it.

**Comment by phillipsj** [comment]
Thanks for clarifying.  I will try to get those changes made this weekend.

**Comment by sindresorhus** [comment]
ping @phillipsj

**Comment by phillipsj** [comment]
Sorry life is throwing curveballs at me.  I have a few more commits to
squash and I will push it.

**Pull request closed without merging**